Apply uppercase in translation (closes #4)

NicolasGrosjean opened the pull request:
There are a lot of sentences which are only few words with uppercase at the first letter. We apply this rule to the translation if the original text has that.

NicolasGrosjean committed:
Apply uppercase in translation

There are a lot of sentences which are only few words with uppercase at the first letter.
We apply this rule to the translation if the original text has that.

Jeydolen commented:
Seems alright in terms of implementation but what you are asking depends on languages used.

[English capitalization rules](https://www.grammarly.com/blog/capitalization-rules/)
[French capitalization rules](https://blogs.transparent.com/french/french-rules-of-capitalization/)

We should provide a way for users to enable or not this behaviour (via `process.env` for example)

NicolasGrosjean committed:
Configure deepl post processing

Disable it by default to keep default current behaviour

NicolasGrosjean commented:
> Seems alright in terms of implementation but what you are asking depends on languages used.
> 
> [English capitalization rules](https://www.grammarly.com/blog/capitalization-rules/) [French capitalization rules](https://blogs.transparent.com/french/french-rules-of-capitalization/)
> 
> We should provide a way for users to enable or not this behaviour (via `process.env` for example)

Thanks for the links.

If I understand correctly, the use case is the rule for titles (events, modifiers, focus, ...).

I added a config entry, but if you prefer I use another configuration file, I can.

NicolasGrosjean committed:
Manage capitalize exception in English

Some words are not capitalized in titles

NicolasGrosjean committed:
Manage capitalize exceptions in French

Some words are not capitalized in titles and we capitalize words after a single quote

NicolasGrosjean committed:
Fix 46fedd0

Jeydolen committed:
Update helper.js

Added startup error handling for post processing

NicolasGrosjean committed:
Add more capital word exceptions
